netfx_loader/ClrLoader.cs

```diff
@@ -77,21 +74,38 @@ public static void CloseAppDomain(IntPtr domain)
         {
             if (domain != IntPtr.Zero)
             {
-                var handle = GCHandle.FromIntPtr(domain);
-                var domainObj = (AppDomain)handle.Target;
-                AppDomain.Unload(domainObj);
-                handle.Free();
+                try
+                {
+                    var domainData = _domains[(int)domain];
+                    domainData.Dispose();
+                }
+                catch (Exception exc)
+                {
+                    Print($"Exception in {nameof(CloseAppDomain)}: {exc.GetType().Name} {exc.Message}\n{exc.StackTrace}");
+                }
+            }
+        }
+
+        [DllExport("pyclr_finalize", CallingConvention.Cdecl)]
+        public static void Close()
+        {
+            foreach (var domainData in _domains)
+            {
+                domainData.Dispose();
             }
+
+            _domains.Clear();
+            _initialized = false;
         }
 
-        #if DEBUG
-        static void Print(string s)
+#if DEBUG
+        internal static void Print(string s)
```

Comment:
I wonder why not just use `Conditional("DEBUG")`

Reply:
Yeah, I'll probably do that later, thanks for the hint :)